earthfile2llb/interpreter.go

```diff
@@ -1239,6 +1248,79 @@ func (i *Interpreter) handleBuild(ctx context.Context, cmd spec.Command, async b
 	return nil
 }
 
+func (i *Interpreter) handleWildcardBuilds(ctx context.Context, fullTargetName string, cmd spec.Command, async bool) error {
+
+	parsedTarget, err := domain.ParseTarget(fullTargetName)
+	if err != nil {
+		return i.wrapError(err, cmd.SourceLocation, "failed to parse target")
+	}
+
+	dir, base := filepath.Split(parsedTarget.GetLocalPath())
+	if strings.Contains(dir, "*") || base != "*" {
+		return i.errorf(cmd.SourceLocation, "wildcard BUILD pattern must end with a single '*'")
+	}
+
+	isRemoteTarget := i.target.IsRemote()
+
+	var matches []string
+
+	if isRemoteTarget {
+		matches, err = i.converter.ExpandRemoteWildcard(ctx, i.target, parsedTarget.GetLocalPath())
+	} else {
+		matches, err = fileutil.GlobDirs(parsedTarget.GetLocalPath())
+	}
+	if err != nil {
+		return i.wrapError(err, cmd.SourceLocation, "invalid BUILD wildcard pattern")
```

Comment:
the `"invalid BUILD wildcard pattern"` error message is confusing for cases where `ExpandWildcard` returns an error that's not related to the pattern (e.g. a network-related error)

Reply:
Updated in upcoming commit. The language is now more generic.